### PR TITLE
feat(5d): session recording and playback (.ksr format)

### PR DESCRIPTION
**Pull request by ikari-pl**
## Summary
- Record and replay keyboard input sessions with frame-accurate timing
- `.ksr` binary format embeds a full SNA snapshot for deterministic replay
- SessionRecorder class with record/playback state machine
- Binary event stream: FRAME_SYNC markers + KEY_DOWN/KEY_UP/JOY events
- Portable little-endian serialization (no struct packing)
- Main loop hooks capture keyboard_matrix diffs per frame
- IPC commands: `session record/play/stop/status`
- 10 unit tests covering round-trip, validation, edge cases

## Test plan
- [x] `./test_runner --gtest_filter='SessionRecorder*'` — 10 tests pass
- [x] Full test suite: 599 pass, 2 skipped (same baseline)
- [ ] Manual: record a session, play it back, verify deterministic replay
- [ ] Manual: `echo "session status" | nc localhost 6543`

### CODE REVIEW

File: src/session_recording.cpp [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    if (state_ == SessionState::RECORDING) stop_recording();
+    if (state_ == SessionState::PLAYING) stop_playback();
+}
+
+bool SessionRecorder::start_recording(const std::string& path, const std::string& snap_path) {
+    if (state_ != SessionState::IDLE) return false;
+
+    // Read the SNA file that was saved before calling us
+    FILE* snap_f = fopen(snap_path.c_str(), "rb");
+    if (!snap_f) return false;
```
[the remark this answers is not in the document]
Comment (ikari-pl) [reply src/session_recording.cpp]: Fixed — introduced RAII FilePtr wrapper (unique_ptr<FILE, FileCloser>) for all local FILE* variables. Early returns now auto-close. Member rec_file_ stays raw since it spans method calls.

File: src/session_recording.cpp [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+    // Write header (32 bytes)
+    uint8_t header[KSR_HEADER_SIZE] = {};
+    header[0] = 'K'; header[1] = 'S'; header[2] = 'R'; header[3] = 0x1A;
+    header[4] = KSR_VERSION;
+    // SNA size at offset 8 (LE32)
+    header[8]  = static_cast<uint8_t>(snap_size);
+    header[9]  = static_cast<uint8_t>(snap_size >> 8);
+    header[10] = static_cast<uint8_t>(snap_size >> 16);
+    header[11] = static_cast<uint8_t>(snap_size >> 24);
```
[the remark this answers is not in the document]
Comment (ikari-pl) [reply src/session_recording.cpp]: Fixed — extracted write_le32_buf() helper, used for header encoding instead of manual byte assignment.

File: src/session_recording.cpp [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+        if (evt.type != SessionEventType::FRAME_SYNC) {
+            uint8_t d[2];
+            if (fread(d, 1, 2, f) != 2) break;
+            evt.data = read_le16(d);
+        } else {
+            total_frames_++;
+        }
+        events_.push_back(evt);
+    }
```
[the remark this answers is not in the document]
Comment (ikari-pl) [reply src/session_recording.cpp]: Fixed — replaced while(\!feof()) with idiomatic read-first for(;;) loop.
